src/Navbar/Navbar.js

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import "./styles.scss";
+
+const Link = (props) => {
+  return <li className={`${props.className || ""}`}>
+    <a href={props.to}
+      target={props.target}>
+      {props.name}
+    </a>
+  </li>;
+};
+
+Link.defaultProps = {
+  className: "",
+  to: "#",
+  target: "_self",
+  name: "Link"
+}
+
+class Dropdown extends Component {
+  constructor(props) {
+    super(props);
+    this.activate = this.activate.bind(this);
+    this.state = { dd_className: "dropdown"};
+  }
+ 
+  activate(classes) {
+    this.setState({dd_className: classes.join(" ")});
+  };
+
+  render() {
+    let {state, props} = this;
+    return (
+      <li className={state.dd_className}
+        onMouseEnter={this.activate.bind(null, ["dropdown", "active"])}
+        onMouseLeave={this.activate.bind(null, ["dropdown"])} >
+        <a href="#">{props.name}</a>
+        <div className="dd-container">
+          <ul className="dd-list">
+            {props.submodules.map((submod) => <Link className="dd-element" key={submod.name} {...submod}/>)}
+          </ul>
+        </div>
+      </li>
+    );
+  }
+};
+
+Dropdown.defaultProps = {
+  name: "Dropdown",
+  submodules: [
+    {
+      name: "GUIMDevTeam",
+      to: "https://github.com/Intelimetrica/GUIM",
+      permission: "default"
+    },
+    {
+      name: "Intelimetrica",
+      to: "http://intelimetrica.com",
+      permission: "default"
+    }
+  ]
+}
+
+class Navbar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { permissions: ["default"] };
+    this._buildHeaders();
+  }
+
+  _havePermission(permissions, to_validate){
+    return (permissions.indexOf(to_validate) >= 0) || to_validate === "default";
+  }
+
+  _buildHeaders() {
+    const { modules } = this.props;
+    if(modules.length < 1){
+      return;
+    }
+    return modules.filter((module) => this._havePermission(this.state.permissions, module.permission))
+      .map((module) => {
+        if(module.mobile === true) return;
```

Comment:
you can skip this line if you use Array.filter and then map
.filter(module => !module.mobile)
.map(module => {
});

Reply:
True, looks cleaner this way.